### ISSUE TITLE
Dataproc leads to misleading software executions

### ISSUE THREAD

**Issue by MattWellie** [populationgenomics/production-pipelines#516]
Issue identified where an MT annotation being present caused ES-Index generation to fail
https://batch.hail.populationgenomics.org.au/batches/429560

Problem batch which caused this: https://batch.hail.populationgenomics.org.au/batches/429539

1. Callset generated by seqr-loader pipeline
2. VEP JSON -> Hail tables was run without dataproc (default flag is False on use_dataproc)
3. AnnotateCohort, AnnotateDataset, and MtToEs ran inside dataproc:
  - Dataproc uses a base image australia-southeast1-docker.pkg.dev/analysis-runner/images/dataproc:hail-0.2.105 ( we are on 0.2.118 in prod), and uses the install script gs://cpg-common-main/hail_dataproc/install_common.sh to install updated libraries
      - FYI Hail is not updated, and the Dataproc AnnotateCohort/Dataset logging shows that the Hail version is still on 105 at runtime
  - Almost all the libraries to install in that script have no pinned versions (i.e. if any version is currently installed, then nothing will be updated)
  - AnnotateCohort and AnnotateDataset work by copying the contents of cpg_workflows/query_modules into the dataproc environment. This meant that an earlier (unspecified, but not recent..) version of the production-pipelines codebase was used to run those stages
  - The MtToEs stage doesn't use any pyfiles, and just executes the specified script
  - As a marker, the removal of InbreedingCoeff annotation from the MT prior to exporting to ES Index used to be done in MtToEs, but was moved to AnnotateDataset be compatible with GATK-SV. This was [moved in mid-September](https://github.com/populationgenomics/production-pipelines/commit/08060a5d1f62735ffffb529527f51b24fd885f6e), so the best case scenario here is that Dataproc is running code 1.5 months old.
  - The AnnotateCohort/Dataset stages ran using an OLD version of the production-pipelines codebase, and MtToEs ran with the latest version, so this specific annotation was not removed, crashing the ES export/ingestion.

TL;DR Dataproc is dangerous for our use case - we think we know which code is running, and we are wrong.

**Comment by MattWellie** [comment]
Dataproc is definitely misleading, but DAMN it's fast:
[Dataproc pipeline execution](https://batch.hail.populationgenomics.org.au/batches/429539) (AnnotateCohort took 27 mins)
[Non-Dataproc](https://batch.hail.populationgenomics.org.au/batches/429639/jobs/614) (AnnotateCohort failed after 2 hours)

The Query-On-Batch job doing all the work appears to have failed...
https://batch.hail.populationgenomics.org.au/batches/429653/jobs/61

There are 4 checkpoints created during AnnotateCohort, and the timestamps suggest that it was all going smoothly until a failure, so that duration may be completely misleading
- mt-vep-split.mt: 2023-11-09T04:06:09Z
- vqsr.ht: 2023-11-09T04:12:11Z
- mt-vep-split-vqsr.mt: 2023-11-09T04:14:01Z

**Comment by MattWellie** [comment]
Restarted here
https://batch.hail.populationgenomics.org.au/batches/429657
